public/app/core/components/TransformersUI/FilterByValueTransformerEditor.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useMemo } from 'react';
+import {
+  DataTransformerID,
+  standardTransformers,
+  TransformerRegistyItem,
+  TransformerUIProps,
+  getFieldDisplayName,
+  Field,
+  DataFrame,
+} from '@grafana/data';
+import { getAllFieldNamesFromDataFrames } from './OrganizeFieldsTransformerEditor';
+import { Select, Button, Input } from '@grafana/ui';
+
+import {
+  FilterByValueTransformerOptions,
+  ValueFilter,
+} from '@grafana/data/src/transformations/transformers/filterByValue';
+
+import { valueFiltersRegistry, ValueFilterID } from '@grafana/data/src/transformations/valueFilters';
+
+function FilterSelectorRow(props: any) {
+  const { fieldNameOptions, onDelete, onConfigChange, config, fieldType } = props;
+
+  console.log('props', props);
+
+  // Find filter types that fit the chosen field type
+  let filterTypeOptions = valueFiltersRegistry
+    .list()
+    .filter(element => {
+      if (element.supportedFieldTypes) {
+        return !element.supportedFieldTypes ? true : element.supportedFieldTypes.includes(fieldType);
+      } else {
+        return true;
+      }
+    })
+    .map(item => ({ value: item.id, label: item.name, description: item.description }));
```

Comment:
```suggestion
const filterTypeOptions = useMemo(() => {
    return valueFiltersRegistry
      .list()
      .filter(element => {
        if (!Array.isArray(element?.supportedFieldTypes)) {
          return true;
        }
        return element.supportedFieldTypes.includes(fieldType);
      })
      .map(item => ({
        value: item.id,
        label: item.name,
        description: item.description,
      }));
  }, [fieldType]);
```

If you use the `react hook` called `useMemo` like this we will only recalculate the filterTypeOptions if the `fieldType` change instead of doing it on every render.